state-chain/custom-rpc/src/lib.rs

```diff
@@ -1093,7 +1156,15 @@ where
 	) -> Result<(), SubscriptionEmptyError> {
 		let from = from_asset.try_into().map_err(|_| SubscriptionEmptyError)?;
 		let to = to_asset.try_into().map_err(|_| SubscriptionEmptyError)?;
-		self.new_items_subscription(sink, move |api, hash| api.cf_prewitness_swaps(hash, from, to))
+		self.new_items_subscription(sink, move |api, hash| {
+			api.cf_prewitness_swaps(hash, from, to).map(|res| {
+				res.map(|swaps| RpcPrewitnessedSwap {
+					from_asset,
+					to_asset,
+					amounts: swaps.into_iter().map(|s| s.into()).collect(),
+				})
+			})
+		})
 	}
 
 	fn cf_prewitness_swaps(
```

Comment:
Also see how the return tpyes of the single requests and subscriptions for price match, I think we should probably do the same here, and make cf_prewitness_swaps return RpcPrewitnessedSwap.

Also as a note, the cf_subscribe_prewitness_swaps and cf_prewitness_swaps interfaces can be changed with worrying about backcompat as we know no one is using them. Unlikely all the other rpcs, so I'm fine to not introduce new v2 rpcs here for the prewitness rpcs.